hudi-client/src/main/java/org/apache/hudi/client/utils/NetUtils.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.utils;
+
+import org.apache.hudi.exception.HoodieException;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+
+/**
+ * Utility for various network related tasks (such as finding free ports).
+ */
+public class NetUtils {
+  // ------------------------------------------------------------------------
+  //  Encoding of IP addresses for URLs
+  // ------------------------------------------------------------------------
+
+  /**
+   * Encodes an IP address properly as a URL string. This method makes sure that IPv6 addresses have
+   * the proper formatting to be included in URLs.
+   *
+   * @param address The IP address to encode.
+   * @return The proper URL string encoded IP address.
+   */
+  public static String ipAddressToUrlString(InetAddress address) {
```

Comment:
Would like to make sure, the methods here are not copied from someplace else.. I see a similar class in Flink. We have attribute code if we reuse from another project. This builds overhead when we release in terms of license clearance.. Please respond, so we can decide accordingly

Reply:
Yes, it is copyed from Apache FLINK, seems like to implement it ourself.